src/hotspot/share/oops/instanceKlass.hpp

```diff
@@ -543,7 +543,8 @@ class InstanceKlass: public Klass {
   bool is_not_initialized() const          { return init_state() <  being_initialized; }
   bool is_being_initialized() const        { return init_state() == being_initialized; }
   bool is_in_error_state() const           { return init_state() == initialization_error; }
-  bool is_reentrant(Thread *thread)        { return thread == _init_thread; }
+  bool is_reentrant_initialization(Thread *thread) { return thread == _init_thread; }
+  bool is_reentrant_linking(Thread *thread)        { return thread == _init_thread; }
```

Comment:
Sorry Coleen when I asked for this I thought these queries included the state check, but now I see the state is checked separately. You may as well just have a single simple query `is_init_thread(JavaThread* current)`.
And wherever we set the `_init_thread` we should have an assert that it is NULL (if not already done). Thanks.

Reply:
Ok, your suggested name change is good and I added the assert we talked about. JCK vm tests pass but I'm rerunning tier1-4 on this change.